integration/single-node-slug-deploy/check.go

```diff
@@ -333,21 +373,81 @@ func postHelloManifest(dir string) error {
 
 	f, err := os.OpenFile(manifestPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
 	if err != nil {
-		return err
+		return fields.ID(""), err
 	}
 	defer f.Close()
 	err = manifest.Write(f)
 	if err != nil {
-		return err
+		return fields.ID(""), err
 	}
 	f.Close()
 
 	manifestPath, err = signManifest(manifestPath, dir)
 	if err != nil {
-		return err
+		return fields.ID(""), err
+	}
+
+	cmd := exec.Command("p2-rctl", "--log-json", "create", "--manifest", manifestPath, "--node-selector", "test=yes")
+	out := bytes.Buffer{}
+	cmd.Stdout = &out
+	cmd.Stderr = &out
+	err = cmd.Run()
+	if err != nil {
+		return fields.ID(""), fmt.Errorf("Couldn't create replication controller for hello: %s %s", out.String(), err)
+	}
+	var rctlOut struct {
+		ID string `json:"id"`
+	}
+
+	err = json.Unmarshal(out.Bytes(), &rctlOut)
```

Comment:
you did it better than I did (see my code that just parses the UUID out of the message with a regex...)